tests/integration/distribution.go

```diff
@@ -147,15 +148,20 @@ func (s *CCVTestSuite) TestRewardsDistribution() {
 		1,
 	)
 
-	// Check that the consumer rewards allocation are empty since relayAllCommittedPackets calls BeginBlockRD,
-	// which in turns calls AllocateTokens.
+	// Consumer allocations are distributed between the validators and the community pool.
+	// The decimals resulting from the distribution are expected to remain in the consumer allocations.
 	rewardsAlloc := providerKeeper.GetConsumerRewardsAllocation(s.providerCtx(), s.consumerChain.ChainID)
-	s.Require().Empty(rewardsAlloc.Rewards)
+	remainingAlloc := rewardsAlloc.Rewards.AmountOf(rewardsIBCdenom)
+	s.Require().True(remainingAlloc.LTE(math.LegacyOneDec()))
```

Comment:
Why `LTE` and not `LT` (i.e., less than)? It seems that if `remainingAlloc` was `1` it should have been allocated?

Reply:
It's due to `SendCoinsFromModuleToModule` that only accepts integer amounts of coins. Therefore when the total reward is split in two, the truncation remainders of the [validator allocation](https://github.com/cosmos/interchain-security/blob/f031dafc5b889899d8f5ee8e862d0b235bd8ed81/x/ccv/provider/keeper/distribution.go#L130) and the [community pool transfer](https://github.com/cosmos/interchain-security/blob/f031dafc5b889899d8f5ee8e862d0b235bd8ed81/x/ccv/provider/keeper/distribution.go#L149) can sum to one. See [here](https://github.com/cosmos/interchain-security/blob/f031dafc5b889899d8f5ee8e862d0b235bd8ed81/x/ccv/provider/keeper/distribution.go#L161-L162) how the consumer rewards value gets updated.